lib/AST/Attr.cpp

```diff
@@ -970,9 +970,9 @@ bool DeclAttribute::printImpl(ASTPrinter &Printer, const PrintOptions &Options,
     break;
   }
   case DAK_Custom: {
-
     auto attr = cast<CustomAttr>(this);
-    if (auto type = attr->getType()) {
+    if (auto type =
+            D->getResolvedCustomAttrType(const_cast<CustomAttr *>(attr))) {
```

Comment:
Should getType() be made private?

Reply:
There are a lot of existing uses of `getType()` that don't need to go through something like `getResolvedCustomAttrType()` because they get called on a `CustomAttr` that was retrieved via a request that resolves the type. It's only when printing or serializing that we work with `CustomAttrs` abstractly (without knowing that we want a result builder, property wrapper, etc.) and therefore need this.